packages/frontend/src/components/accounts/two_factor/TwoFactorVerifyModal.js

```diff
@@ -1,32 +1,125 @@
+    },
+} = utils;
+
 const Form = styled.form`
     display: flex;
     flex-direction: column;
     align-items: center;
     width: 100%;
 `;
 
+const formatNear = (amount) => {
+    try {
+        return formatNearAmount(amount, 4);
+    } catch {
+        // Near amount may be stored as hex rather than decimal
+        return formatNearAmount(new BN(amount, 16).toString(), 4);
+    }
+};
+
+const getTranslationsFromMultisigRequest = ({ actions, receiverId, accountId }) => {
+    const fullAccessKeyAction = actions.find(({ enum: type, permission }) => type === 'addKey' && !permission);
+    if (fullAccessKeyAction) {
+        const publicKey = new PublicKey({
+            keyType: KeyType.ED25519,
+            data: fullAccessKeyAction.addKey.publicKey.data
+        });
+        return [
+            {
+                id: 'twoFactor.action.addKey.full',
+                data: {
+                    accountId,
+                    publicKey: publicKey.toString(),
+                }
+            }
+        ];
+    }
+
+    return actions
+        .map(({ enum: actionType, [actionType]: action }) => {
+            switch (actionType) {
+                case 'addKey':
+                    return {
+                        id: 'twoFactor.action.addKey.limited',
+                        data: {
+                            receiverId,
+                            methodNames: action.permission.functionCall.methodNames.join(', '),
+                            allowance: formatNearAmount(action.permission.functionCall.allowance, 4),
+                            publicKey: new PublicKey({ keyType: KeyType.ED25519, data: action.publicKey.data }).toString(),
+                        }
+                    };
+                case 'deleteKey':
+                    return {
+                        id: 'twoFactor.action.deleteKey',
+                        data: {
+                            publicKey: new PublicKey({ keyType: KeyType.ED25519, data: action.publicKey.data }).toString(),
+                        }
+                    };
+                case 'functionCall':
+                    return {
+                        id: 'twoFactor.action.functionCall',
+                        data: {
+                            receiverId,
+                            methodName: action.methodName,
+                            deposit: formatNear(action.deposit, 4),
+                            args: Buffer.from(action.args).toString(),
+                        }
+                    };
+                case 'transfer':
+                    return {
+                        id: 'twoFactor.action.transfer',
+                        data:{
+                            receiverId,
+                            deposit: formatNearAmount(action.deposit, 4),
+                        }
+                    };
+                case 'stake':
+                    return {
+                        id: 'twoFactor.action.stake',
+                        data: {
```

Comment:
This matches what we have in `near-contract-helper` but doesn't line up with what i've found in the [docs](https://nomicon.io/RuntimeSpec/Actions#stakeaction). Staking is actually achieved through calling a function on the validator contract (`functionCall`), it seems this is more related to validators themselves, is it actually possible to trigger this action from the wallet?

Reply:
Hmm good catch. I think `stake` might be referring to actual staking, not delegating to a validator that then stakes on behalf of the user. Actual staking is not possible in the wallet at the moment.

Reply:
Yea probs worth removing since it won't be used here if we add support anyway